lib/runner.js

```diff
@@ -386,6 +386,15 @@ Runner.prototype.hook = function(name, fn) {
       }
       if (err) {
         if (err instanceof Pending) {
+          if (name === HOOK_TYPE_AFTER_EACH || name === HOOK_TYPE_AFTER_ALL) {
+            utils.deprecate(
+              util.format(
+                "skipping a test within an '%s' hook is DEPRECATED and will throw an exception in a future version of Mocha. " +
+                  'Use a return statement or other means to abort hook execution.',
+                name === HOOK_TYPE_AFTER_EACH ? 'afterEach()' : 'after()'
```

Comment:
```js
               sQuote(name === HOOK_TYPE_AFTER_ALL ? 'after' : name)
```

Reply:
Why don't the expected names match? Let's fix **"lib/suite.js"** to fix _this_...
* `HOOK_TYPE_AFTER_ALL: 'after'`
* `HOOK_TYPE_BEFORE_ALL: 'before'`

Reply:
Or:
```js
function name2HookType(name) {
  var hookTypeFor = {};
  hookTypeFor[HOOK_TYPE_AFTER_ALL] = 'after';
  hookTypeFor[HOOK_TYPE_AFTER_EACH] = HOOK_TYPE_AFTER_EACH;
  return hookTypeFor[name];
}
...
               sQuote(name2HookType(name)),
...
};


```